src/features/Chat/components/ActiveWindow/Dialogs/ReportDialog.tsx

```diff
@@ -0,0 +1,147 @@
+import { selectUserId } from '@/features/Authentication/userSlice';
+import { useAppSelector } from '@/store';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import styled from 'styled-components';
+import { IconButton, TextButton } from '@/components/Buttons';
+import LabeledInput from '@/components/LabeledInput';
+import { palette } from '@/components/variables';
+import ReportSuccessDialog from './ReportSuccessDialog';
+import Text from '@/components/Text';
+
+type Props = {
+  buddyId: string;
+  close: () => void;
+};
+
+const ReportDialog = ({ buddyId, close }: Props) => {
+  const { t } = useTranslation('chat');
+
+  const userId = useAppSelector(selectUserId);
+
+  const sendReportRequest = async () => {
+    fetch('/api/report', {
+      method: 'POST',
+      body: JSON.stringify({
+        contact_field: contactInfo,
+        report_reason: reportReason,
+        reported_user_id: buddyId,
+        reporter_user_id: userId,
+      }),
+    })
+      .then(response => {
+        if (response.status == 200) setIsReported(true);
+      })
+      .catch(error => {
+        console.error(error);
+      });
+  };
```

Comment:
This is a no go route for me. We are using redux-tookit for the project to handle these async actions. Please use the [rtk-mutation](https://redux-toolkit.js.org/rtk-query/usage/mutations) for this.

Reply:
An example could be sending a message:
https://github.com/sos-lapsikyla/ylitse-web/blob/master/src/features/Chat/chatPageApi.ts
https://github.com/sos-lapsikyla/ylitse-web/blob/master/src/features/Chat/components/ActiveWindow/index.tsx

Reply:
This was for testing purposes, but I haven't been able to test the API, because the code isn't merged yet. 🤔

Reply:
Could you run the API using the feature-branch that has the reporting-code? That way you can make sure it works =)

Reply:
https://gitlab.com/ylitse/ylitse-api/-/merge_requests/67
You need to add the feature flag in the Makefile if I remmeber correctly. Unfortunately it was not documented in the MR